app/schemas.py

```diff
@@ -178,25 +180,116 @@ class ServiceSchema(BaseSchema):
     organisation = field_for(models.Service, 'organisation')
     branding = field_for(models.Service, 'branding')
     dvla_organisation = field_for(models.Service, 'dvla_organisation')
+    permissions = fields.Method("service_permissions")
+    override_flag = False
+
+    def service_permissions(self, service):
+        permissions = []
+        str_permissions = []
+
+        perms = dao_fetch_service_permissions(service.id)
+        for p in perms:
+            permission = {
+                "service_id": service.id,
+                "permission": p.permission
+            }
+            permissions.append(permission)
+            str_permissions.append(p.permission)
+
+        def deprecate_convert_flags_to_permissions():
+            def convert_flags(flag, notify_type):
+                if flag and notify_type not in str_permissions:
+                    permission = {
+                        "service_id": service.id,
+                        "permission": notify_type
+                    }
+                    permissions.append(permission)
+                elif flag is False and notify_type in str_permissions:
+                    permission = {
+                        "service_id": service.id,
+                        "permission": notify_type
+                    }
+                    permissions.remove(permission)
+
+            convert_flags(service.can_send_international_sms, INTERNATIONAL_SMS_TYPE)
+            convert_flags(service.can_send_letters, LETTER_TYPE)
+
+        deprecate_convert_flags_to_permissions()
+
+        return permissions
 
     class Meta:
         model = models.Service
-        exclude = ('updated_at',
-                   'created_at',
-                   'api_keys',
-                   'templates',
-                   'jobs',
-                   'old_id',
-                   'template_statistics',
-                   'service_provider_stats',
-                   'service_notification_stats')
+        exclude = (
+            'updated_at',
+            'created_at',
+            'api_keys',
+            'templates',
+            'jobs',
+            'old_id',
+            'template_statistics',
+            'service_provider_stats',
+            'service_notification_stats',
+        )
         strict = True
 
     @validates('sms_sender')
     def validate_sms_sender(self, value):
         if value and not re.match(r'^[a-zA-Z0-9\s]+$', value):
             raise ValidationError('Only alphanumeric characters allowed')
 
+    @validates('permissions')
+    def validate_permissions(self, value):
+        permissions = [v.permission for v in value]
+        for p in permissions:
+            if p not in models.SERVICE_PERMISSION_TYPES:
```

Comment:
I thought it would be good to do it here to avoid hitting the database and raise a nicer error